.github/workflows/scripts/e2e-maven-push.sh

```diff
@@ -0,0 +1,152 @@
+version_major() {
+    VER=$(echo $1 | cut -d '.' -f1)
+    echo "$VER"
+}
+
+# version_minor prints the minor version number.
+# Expects a string like '1.19.7'.
+# version_minor returns "19" if the input is '1.19.7'
+version_minor() {
+    VER=$(echo $1 | cut -d '.' -f2)
+    echo "$VER"
+}
+
+# version_patch prints the patch version number.
+# Expects a string like '1.19.7-SNAPSHOT.jar'
+# version_patch returns "7" if the input is '1.19.7'
+version_patch() {
+    VER=$(echo $1 | cut -d '.' -f3)
+    echo "$VER"
+}
+
+# Bumps the version
+new_version() {
+    current_tag=$1
+    release_major=$(version_major "$current_tag")
+    release_minor=$(version_minor "$current_tag")
+    release_patch=$(version_patch "$current_tag")
+
+    # These if-statements are sorted by likelihood
+    if [[ $release_patch != "99" ]]; then
+        # Only need to bump the patch
+        release_patch=$((release_patch+1))
+    elif [[ $release_patch = "99" && $release_minor != "99" ]]; then
+        # Need to bump minor
+        release_minor=$(($release_minor+1))
+        release_patch="0"
+    elif [[ $release_patch = "99" && $release_minor = "99" ]]; then
+        # Need to bump major
+        release_major=$(($release_major+1))
+        release_minor="0"
+        release_patch="0"
+    fi
+    echo $release_major.$release_minor.$release_patch
+}
+
+next_tag=$(new_version $artifact_tag)
+
+# Output the artifact name
+echo "artifact-version=${artifact_tag}" >> $GITHUB_OUTPUT
+
+tag=$(mvn versions:set -DnewVersion=$next_tag)
+cd -
+
+# Commit the new version.
+git commit -m "${GITHUB_WORKFLOW}" "${package_dir}/pom.xml" "${package_dir}/pom.xml"
+
+# If this is an e2e test for a tag, then tag the commit and push it.
+this_event=$(e2e_this_event)
+echo "this_event: ${this_event}"
+if [ "${this_event}" == "tag" ] || [ "${this_event}" == "create" ]; then
+    git tag "${tag}"
+fi
+
+git remote -v
+git branch
+pwd
+if [ "${branch}" != "main" ]; then
+    # Reset branch1 and push the new version.
+    # git branch -D "$branch"
+    git checkout -b "$branch"
+    if [ "${this_event}" == "tag" ] || [ "${this_event}" == "create" ]; then
+        git push --set-upstream origin "${branch}" "${tag}" -f
+    else
+        git push --set-upstream origin "$branch" -f
+    fi
+    git checkout main
+
+    # Update a dummy file to avoid https://github.com/slsa-framework/example-package/issues/44
+    date >./e2e/dummy
+    git add ./e2e/dummy
+    git commit -m "sync'ing branch1 - $(cat ./e2e/dummy)"
+    git push origin main
+else
+    if [ "${this_event}" == "tag" ] || [ "${this_event}" == "create" ]; then
+        # TODO(#213): push tag separately until bug is fixed.
+        # NOTE: If there is a concurrent update to main we want it to fail here
+        # without pushing the tag because we will lose the changes to main.
+        git push origin main
+        git push origin "${tag}"
+    else
+        git push origin maven-e2e-temp # TODO: CHANGE to main!!!!!!!!!!
```

Comment:
Add a issue # to the TODO if you intend to merge it.

Reply:
We should change it before merging this PR.